app/src/main/java/org/schulcloud/mobile/models/file/FileRepository.kt

```diff
@@ -2,42 +2,50 @@ package org.schulcloud.mobile.models.file
 
 import androidx.lifecycle.LiveData
 import io.realm.Realm
-import org.schulcloud.mobile.jobs.ListDirectoryContentsJob
+import org.schulcloud.mobile.jobs.base.RequestJob
 import org.schulcloud.mobile.models.base.Repository
 import org.schulcloud.mobile.models.user.UserRepository
 import org.schulcloud.mobile.utils.*
 
 object FileRepository : Repository() {
     const val CONTEXT_MY = "my"
-    const val CONTEXT_MY_API = "users"
-    const val CONTEXT_COURSES = "courses"
+    const val CONTEXT_MY_API = "user"
+    const val CONTEXT_COURSE = "course"
 
-    fun files(realm: Realm, path: String): LiveData<List<File>> {
-        return realm.fileDao().files(path)
-    }
+    val user: String
+        get() = UserRepository.userId!!
 
-    fun directories(realm: Realm, path: String): LiveData<List<Directory>> {
-        return realm.fileDao().directories(path)
+    fun files(realm: Realm, owner: String, parent: String?): LiveData<List<File>> {
+        return realm.fileDao().files(owner, parent)
     }
 
-
-    suspend fun syncDirectory(path: String) {
-        ListDirectoryContentsJob(path).run()
+    fun directories(realm: Realm, owner: String, parent: String?): LiveData<List<File>> {
+        return realm.fileDao().directories(owner, parent)
     }
 
-
-    fun fixPath(path: String): String {
-        var fixedPath = path.trimLeadingSlash().ensureTrailingSlash()
-        if (path.startsWith(CONTEXT_MY))
-            fixedPath = path.replaceRange(0, CONTEXT_MY.length, pathPersonal("").trimTrailingSlash())
-        return fixedPath
+    fun directory(realm: Realm, id: String): File? {
+        return realm.fileDao().directory(id)
     }
 
-    fun pathPersonal(path: String? = null): String {
-        return combinePath(CONTEXT_MY_API, UserRepository.userId!!, path)
+    suspend fun syncDirectory(owner: String, parent: String?) {
+        RequestJob.Data.with({ listDirectoryContents(owner, parent) }, {
+                beginGroup()
+                        .equalTo("isDirectory", false)
+                        .equalTo("owner", owner)
+                        .or()
+                        .notEqualTo("owner", owner)
+                        .endGroup()
```

Comment:
Are you sure that this selection of deprecated elements to delete is correct? Shouldn't it just select by parent?